### DISCUSSION TITLE
Problems building without --disable-openfec --disable-tools

### DISCUSSION THREAD

**Discussion by DColadas** [roc-streaming/roc-toolkit#349] [Q&A] [answered]
Hello,
I am interested in collaborating in this project. I am trying to get familiar with some of the commands displayed in the docs to check how the program works.
I had no problems installing using the commands:
```
$ scons -Q --enable-pulseaudio-modules --build-3rdparty=openfec,pulseaudio
$ sudo scons -Q --enable-pulseaudio-modules --build-3rdparty=openfec,pulseaudio install
```
as shown in the user cookbook. However, while trying to build the sources with any command of this type:
```
$ scons -Q
$ scons -Q --enable-werror --enable-debug --sanitizers=all
```
I get the error `error: openfec not found (see 'config.log' for details)`. If I add the flag `--disable-openfec`, what I get is a linker error:
```
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libpulse.so: undefined reference to `pa_dbus_wrap_connection_free'
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libpulse.so: undefined reference to `pa_dbus_add_matches'
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libpulse.so: undefined reference to `pa_dbus_wrap_connection_get'
/usr/lib/gcc/x86_64-linux-gnu/7/../../../x86_64-linux-gnu/libpulse.so: undefined reference to `pa_dbus_wrap_connection_new'
collect2: error: ld returned 1 exit status
```
I can solve this problem also adding the `--disable-tools` flag. While it works, I suppose disabling the support for OpenFEC and the tools building may cause some issues while developing. Is there any area where I could contribute to the code without these characteristics enabled?
All this happened in both the `master` and `develop` branches, and on two different Ubuntu 18.04 systems (one of them freshly installed). Any help is appreciated!

**Comment by DColadas** [comment, marked as answer]
Problem solved! I added the `--enable-pulseaudio-modules --build-3rdparty=all` options when building.
Therefore, I am running `scons -Q --enable-werror --enable-debug --sanitizers=all --enable-pulseaudio-modules --build-3rdparty=all` in order to not get the error. 
Hopefully I will be contributing to the project soon!

**Comment by gavv** [comment]
Hi, yes, you need either:

* disable OpenFEC support using `--disable-openfec`
* ask Roc to build OpenFEC for you: `--build-3rdparty=...,openfec,...` or `--build-3rdparty=all`
* download, build, and install OpenFEC manually and let Roc to find it on the system (no special options are needed)

The same is true for PulseAudio, you need either:

* completely disable command-line tools using `--disable-tools` and also don't use `--enable-pulseaudio-modules`
* disable PulseAudio support in tools using `--disable-pulseaudio` and also don't use `--enable-pulseaudio-modules`
* ask Roc to build PulseAudio for you: `--build-3rdparty=...,pulseaudio,...` or `--build-3rdparty=all`
* install PulseAudio headers and libraries system-wide and let Roc to find and use them

  * but if you also want to use `--enable-pulseaudio-modules`, just PulseAudio headers and libraries are not enough, you need complete PulseAudio source tree; in this case you can either use `--build-3rdparty` or build PulseAudio from sources manually and use `--with-pulseaudio`

**Reply by 8julie** [reply]
I had the same issue as well. I don't believe that I need to use OpenFEC so I disabled it with `--disable-openfec` and it works. Thank you guys!
